### PR TITLE
Add new GitHub actions publishing workflow, removes .travis.yml

### PR DESCRIPTION
**Pull request by aireilly**
Ensure that the GitHub Pages site is published from gh-pages branch.

Preview: https://aireilly.github.io/supplementary-style-guide/

image

### CODE REVIEW

File: .github/workflows/deploy-site.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      - name: Install AsciiDoctor
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y asciidoctor
+
+      - name: Build AsciiDoc
+        run: |
+          asciidoctor supplementary_style_guide/main.adoc -o index.html
+          mkdir docs
+          mv index.html docs/
```
Comment (bergerhoffer) [inline .github/workflows/deploy-site.yml:25]: We'll need to add something here to copy the images/ folder too

**Review by bergerhoffer — COMMENTED** [review]
@aireilly you are awesome, thank you so much!

I'll take a closer look through, but this is the first thing I noticed

**Review by bergerhoffer — DISMISSED** [review]
I won't really know til we merge it and test it, but this LGTM for merging. Thanks again!

**Comment by mportman12** [comment]
Is there a way to do a staged merge without taking it to production?

File: .github/workflows/deploy-site.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        run: |
+          asciidoctor supplementary_style_guide/main.adoc -o index.html
+          mkdir docs
+          cp -r supplementary_style_guide/images/ docs/images
+          mv index.html docs/
+
+      - name: Publish to GitHub Pages
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (bergerhoffer) [inline .github/workflows/deploy-site.yml:31]: @aireilly is this GITHUB_TOKEN something we have to configure somewhere?

**Comment by bburt-rh** [comment]
Is there a reason why the `[discrete]` code is being added to `s.adoc`? Otherwise this LGTM, though I also have the question mentioned by @bergerhoffer re the GH token.

**Comment by mportman12** [comment]
> Is there a reason why the `[discrete]` code is being added to `s.adoc`? Otherwise this LGTM, though I also have the question mentioned by @bergerhoffer re the GH token.

@bburt-rh I forgot to add [discrete] when I added the sticky glossary entries, and it didn't get caught in review.

**Comment by bergerhoffer** [comment]
> Is there a way to do a staged merge without taking it to production?

@mportman12 Aidan did share this with me, his test for his own fork: https://aireilly.github.io/supplementary-style-guide/

So that was good enough for me. Plus we _hopefully_ can't mess this up too bad since it's not updating at all currently. Though I will go ahead and make a backup of the gh-pages branch right now just in case - so we can restore if that gets messed up during this testing.

**Review by bburt-rh — DISMISSED** [review]

**Comment by bburt-rh** [comment]
Approved

File: .github/workflows/deploy-site.yml (hunk 2 above)
Comment (aireilly) [reply .github/workflows/deploy-site.yml:31]: Should be available by default.

Comment (bergerhoffer) [reply .github/workflows/deploy-site.yml:31]: FYI for anyone else curious, it's this: https://docs.github.com/en/actions/security-guides/automatic-token-authentication

**Comment by bergerhoffer** [comment]
Unless there are any objections, will plan to merge this tomorrow morning (eastern time) and see how it goes! :crossed_fingers:

**Review by dfitzmau — DISMISSED** [review]

**Review by bredamc — APPROVED** [review]
Looks good enough to try a test! (Caveat that I don't fully understand the code.)

**Comment by bergerhoffer** [comment]
Merging to test and see how this goes!

**Pull request merged**